### DISCUSSION TITLE
How to get the autocompletion for noice commandline?

### DISCUSSION THREAD

**Discussion by thepenguinthatwants** [crivotz/nv-ide#37] [Q&A] [answered]
Hi,
I am trying to understand and implement more custom configuration based on your setup.
How did u manage to get autocompletion to automatically show examples without tab completion and in correct location?
I copied the whole noice.lua file but it didn't seem to be from that.

https://github.com/crivotz/nv-ide/blob/master/screenshots/nv-ide_screenshot_5.png

How was this implemented?

**Comment by crivotz** [comment, marked as answer]
Hi, it is made using https://github.com/hrsh7th/cmp-cmdline.
The configuration can be found here:
https://github.com/crivotz/nv-ide/blob/e831009bbb9339cb54044fff22ecdb536f612c0e/lua/config/completion.lua#L108

**Comment by thepenguinthatwants** [comment]
Thanks for the reply,
I tried to source that file but didn't make any changes. I am also not specifying any native_menu on my configs.

I assume it's not as simple as sourcing this file and installing hrsh7th/cmp-cmdline, hrsh7th/nvim-cmp and hrsh7th/cmp-nvim-lsp.

**Comment by thepenguinthatwants** [comment]
Oh my gotten it to work with adding lpskind and nvim-autopair plugins  :D thanks this is awesome.
